File: src/main/java/com/att/training/katas/calculator/StringCalculator.java
package com.att.training.katas.calculator;

import java.util.stream.Stream;

class StringCalculator {
    
    int add(String input) {
        if (input.isEmpty()) {
            return 0;
        }
        var tokens = tokenize(input);
        validate(tokens);
        return Stream.of(tokens)
                     .mapToInt(Integer::parseInt)
                     .sum();
    }

    private String[] tokenize(String input) {
        return input.split("[,\\n]");
    }

    private void validate(String[] tokens) {
        var negatives = Stream.of(tokens)
                .filter(token -> token.startsWith("-"))
                .toList();

        if (!negatives.isEmpty()) {
            var message = "Negatives not allowed: " + negatives;
            throw new IllegalArgumentException(message);
        }
    }
}

File: src/main/kotlin/calculator/StringCalculator.kt
package com.att.training.katas.kt.calculator

class StringCalculator {
    fun add(input: String): Int {
        if (input.isEmpty()) return 0

        val tokens = input.split(',', '\n')
        validate(tokens)
        return tokens.sumOf { it.toInt() }
    }

    private fun validate(tokens: List<String>) {
        val negatives = tokens.filter { it.startsWith("-") }
        if (negatives.isNotEmpty()) {
            throw IllegalArgumentException("Negatives not allowed: $negatives")
        }
    }
}
